geth/signal/signals.go

```diff
@@ -37,9 +37,40 @@ type Envelope struct {
 	Event interface{} `json:"event"`
 }
 
+// MarshalJSON implements the json.Marshaller interface.
+func (e *Envelope) MarshalJSON() ([]byte, error) {
```

Comment:
could you please explain or give an example when it is necessary

Reply:
Maybe I'm totally off here, but `NodeCrashEvent` is always assigned to an `interface{}` type  field of the `Envelope` struct. If this method is not added, when we call `json.marshal` on `Envelope`, it will not call the newly created `MarshalJSON` function for `NodeCrashEvent` even though the `json.Marshaller` interface is defined on `NodeCrashEvent`.

This PR also assumes that we will always marshal `NodeCrashEvent` to json when we call `json.marshal` on the `Envelope` type (as shown in the tests) and never by directly calling `json.marshal` directly on `NodeCrashEvent`. If the opposite is true, and we will never call `json.marshal` on `Envelope`, then we can get rid of this method entirely.

Reply:
the MarshalJSON method is defined for a pointer, this way it will be called **only** if Event is a pointer. If MarshalJSON method will be defined for value - then it will be called both on a pointer and a value.

https://github.com/dshulyak/status-go/commit/5ab9534578be3bf670e7b68d52a6f0aeda78b464

Reply:
Ah! Of course. That makes things simpler. Updating this in a sec.